### PR TITLE
Add the WebAuthn user ID to LocalAuthSecrets

### PR DESCRIPTION
**Pull request by codingllama**
WebAuthn recommends using a random user handle (aka user ID) in its protocol.
We are adding the user ID to LocalAuthSecrets (under Webauthn) and allowing it
to be updated and read, both through the user and independently.

The lib/auth/webauthn package is going to be responsible for assigning IDs to
users (and using them).

See https://www.w3.org/TR/webauthn-2/#sctn-user-handle-privacy.

WebAuthn Support RFD[1]

[1] https://github.com/gravitational/teleport/pull/7808

### CODE REVIEW

**Comment by codingllama** [comment]
Looks gigantic because of generated protos, but in actuality this one is fairly small.

FYI @zmb3 @quinqu

File: lib/services/local/resource.go [hunk 1]
```diff
@@ -446,6 +446,14 @@ func itemToLocalAuthSecrets(items userItems) (*types.LocalAuthSecrets, error) {
...
+	if items.webauthnLocalAuth != nil {
+		wal := &types.WebauthnLocalAuth{}
+		err := json.Unmarshal(items.webauthnLocalAuth.Value, wal)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		auth.Webauthn = wal
+	}
 
 	// DELETE IN 7.0: these items are migrated to items.mfa on 6.0 first
```
Comment (zmb3) [inline lib/services/local/resource.go:458]: Should we remove this now while you're familiar with this code path?

Comment (codingllama) [reply lib/services/local/resource.go:458]: Yes, a cleanup is on my radar, but I think I'll do it in a separate PR.

**Comment by codingllama** [comment]
Friendly ping @zmb3 and @quinqu ?

**Review by zmb3 — APPROVED** [review]

**Comment by codingllama** [comment]
Rebased on master and solved conflicts with #8012 (essentially by placing WebauthnLocalAuth* methods before WebauthnSessionData* methods - no real issues there).

**Review by quinqu — APPROVED** [review]

**Comment by codingllama** [comment]
Thanks everyone, let the bot dance begin.

**Review by r0mant — APPROVED** [review]

**Pull request merged**